web/ui/react-app/src/pages/targets/ScrapePoolList.tsx

```diff
@@ -1,61 +1,127 @@
-  const targetGroups = groupTargets(activeTargets);
+const kvSearch = new KVSearch({
+  shouldSort: true,
+  indexedKeys: ['labels', 'scrapePool', ['labels', /.*/]],
+});
+
+interface PanelProps {
+  scrapePool: string;
+  targetGroup: ScrapePool;
+  expanded: boolean;
+  toggleExpanded: () => void;
+}
+
+export const ScrapePoolPanel: FC<PanelProps> = (props: PanelProps) => {
+  const modifier = props.targetGroup.upCount < props.targetGroup.targets.length ? 'danger' : 'normal';
+  const id = `pool-${props.scrapePool}`;
+  const anchorProps = {
+    href: `#${id}`,
+    id,
+  };
+  return (
+    <div>
+      <ToggleMoreLess event={props.toggleExpanded} showMore={props.expanded}>
+        <a className={styles[modifier]} {...anchorProps}>
+          {`${props.scrapePool} (${props.targetGroup.upCount}/${props.targetGroup.targets.length} up)`}
+        </a>
+      </ToggleMoreLess>
+      <Collapse isOpen={props.expanded}>
+        <ScrapePoolContent targets={props.targetGroup.targets} />
+      </Collapse>
+    </div>
+  );
+};
+
+// ScrapePoolListContent is taking care of every possible filter
+const ScrapePoolListContent: FC<ScrapePoolListProps> = ({ activeTargets }) => {
+  const initialPoolList = groupTargets(activeTargets);
+  const [poolList, setPoolList] = useState<ScrapePools>(initialPoolList);
   const initialFilter: FilterData = {
     showHealthy: true,
     showUnhealthy: true,
   };
   const [filter, setFilter] = useLocalStorage('targets-page-filter', initialFilter);
 
-  const initialExpanded: Expanded = Object.keys(targetGroups).reduce(
+  const initialExpanded: Expanded = Object.keys(initialPoolList).reduce(
     (acc: { [scrapePool: string]: boolean }, scrapePool: string) => ({
       ...acc,
       [scrapePool]: true,
     }),
     {}
   );
   const [expanded, setExpanded] = useLocalStorage('targets-page-expansion-state', initialExpanded);
-
   const { showHealthy, showUnhealthy } = filter;
+
+  const handleSearchChange = (e: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+    if (e.target.value !== '') {
+      const result = kvSearch.filter(e.target.value.trim(), activeTargets);
+      setPoolList(
+        groupTargets(
+          result.map((value) => {
+            return value.original as unknown as Target;
+          })
+        )
+      );
+    } else {
+      setPoolList(initialPoolList);
+    }
+  };
   return (
     <>
-      <Filter filter={filter} setFilter={setFilter} expanded={expanded} setExpanded={setExpanded} />
-      {Object.keys(targetGroups)
+      <Row xs="4" className="align-items-center">
+        <Col>
+          <Filter filter={filter} setFilter={setFilter} expanded={expanded} setExpanded={setExpanded} />
+        </Col>
+        <Col xs="6">
+          <InputGroup>
+            <InputGroupAddon addonType="prepend">
+              <InputGroupText>{<FontAwesomeIcon icon={faSearch} />}</InputGroupText>
+            </InputGroupAddon>
+            <Input onChange={handleSearchChange} />
```

Comment:
Could you add a placeholder like:

```suggestion
            <Input onChange={handleSearchChange} placeholder="Filter by endpoint or labels"/>
```

Reply:
Maybe also add `autoFocus`?

Reply:
good idea ! Done :)